src/feed/lookup/lookup.ts

```diff
@@ -0,0 +1,44 @@
+import Long from 'long'
+import { Epoch } from './epoch'
+
+export const LowestLevel = 0
+export const HighestLevel = 31
+export const DefaultLevel = HighestLevel
+export const NoClue = new Epoch(0, 0)
+export const worstHint = new Epoch(63, 0)
+
+export class Lookup {
+  getNextLevel(last: Epoch, now: Long): number {
+    let mix = last.base().xor(now)
+    mix = mix.or(Long.fromNumber(1).shiftLeft((last.level as number) - 1))
+
+    if (mix.greaterThan(Long.MAX_UNSIGNED_VALUE.shiftRight(Long.fromNumber(64 - HighestLevel - 1)))) {
```

Comment:
can you elaborate this line please?

Reply:
I can't explain it because it's translated from the original